CONTRIBUTING.md

```diff
@@ -6,6 +6,212 @@ This guide explains how to contribute to cocotb, and documents the processes we
 All processes in this document are designed to streamline the development effort, to avoid bottlenecks, and to ultimately give a pleasant experience to all involved.
 
 
+## Getting Help
+
+Cocotb is a diverse and challenging project to contribute to.
+If you ever feel lost, out of your depth, or simply want to know more, the [cocotb gitter channel](https://gitter.im/cocotb/Lobby) is actively watched by many cocotb users, contributors, and maintainers.
+It is a good idea if you are unsure whether your problem or question is worthy of a Github Issue to first post it to the gitter channel.
+You may also ask questions in [Github issues](https://github.com/cocotb/cocotb/issues).
+If you don't receive any response on the gitter channel or a Github issue, or you want help privately, you may directly contact a [maintainer](#maintainers).
```

Comment:
```suggestion
If you don't receive any response on the Gitter channel or a Github issue, or you want help privately, you may directly contact a [maintainer](#maintainers).
```
I'm not so sure about the explicit "private help" invitation. Maybe just say something like "Please ring again after a while."?